cmd/gui/gui.go

```diff
@@ -0,0 +1,154 @@
+package gui
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	log "github.com/cihub/seelog"
+	"github.com/gorilla/mux"
+)
+
+var (
+	apiKey   string
+	listener net.Listener
+)
+
+// Message struct is for the JSON messages received from a client POST request
+type Message struct {
+	ReqType string `json:"req_type"`
+	Data    string `json:"data"`
+	Payload string `json:"payload"`
+}
+
+// StopGUIServer closes the connection to the HTTP server
+func StopGUIServer() {
+	if listener != nil {
+		listener.Close()
+	}
+}
+
+// StartGUIServer creates the router and starts the HTTP server
+func StartGUIServer() error {
+	apiKey = config.Datadog.GetString("api_key")
+	port := ":" + config.Datadog.GetString("GUI_port")
+
+	if config.Datadog.GetString("GUI_port") == "-1" {
+		log.Infof("Port -1 specified: not starting the GUI server.")
+		return nil
+	}
+
+	// Instantiate the gorilla/mux router
+	router := mux.NewRouter()
+
+	// Serve the (secured) index page on the default endpoint
+	router.Handle("/", accessAuth(http.FileServer(http.Dir("view/private/"))))
```

Comment:
For  packaging, we shouldn't assume that the various support files are at a constant, relative path.  Make this a variable that we can change (at least at compile time) on a platform-specific basis.  This will make packaging much easier.